test/api/unit/models/group_tasks.test.js

```diff
@@ -233,10 +235,60 @@ describe('Group Task Methods', () => {
           expect(syncedTask.checklist.length).to.equal(0);
           expect(syncedMemberTask.checklist.length).to.equal(0);
         });
+
+        it('removes and reassigns completed and unchecked todos to other assignees', async () => {
+          if (taskType !== 'todo') return;
+
+          const updatedLeader = await User.findOne({ _id: leader._id });
+          const updatedLeadersTasks = await Tasks.Task.find({ _id: { $in: updatedLeader.tasksOrder[`${taskType}s`] } });
+          const syncedTask = find(updatedLeadersTasks, findLinkedTask);
+
+          syncedTask.completed = true;
+          await handleSharedCompletion(syncedTask);
+
+          const updatedMember = await User.findOne({ _id: newMember._id });
+          const updatedMemberTasks = await Tasks.Task.find({ _id: { $in: updatedMember.tasksOrder[`${taskType}s`] } });
+          const syncedMemberTask = find(updatedMemberTasks, findLinkedTask);
+
+          syncedTask.completed = false;
+          await handleSharedCompletion(syncedTask);
+
+          const reUpdatedMember = await User.findOne({ _id: newMember._id });
+          const reUpdatedMemberTasks = await Tasks.Task.find({ _id: { $in: reUpdatedMember.tasksOrder[`${taskType}s`] } });
```

Comment:
.exec()

Reply:
actually is it necessary to load all tasks since we always seem to only need one? can we search by id?